mixins/arrow-keys-mixin.md

```diff
@@ -0,0 +1,54 @@
+# ArrowKeysMixin
+
+Used for managing focus with the arrow keys.
```

Comment:
Perhaps we can reformat a little so the code can fit into view, and use more than one sample to show the usage?    

> The focusable elements can be provided in one of two ways.
>
> If the elements are known up-front and are in the element's local DOM scope, simply add the `d2l-arrowkeys-focusable` class.
```javascript
import { ArrowKeysMixin } from '@brightspace-ui/core/mixins/arrowkeys-mixin.js';
class MyElement extends ArrowKeysMixin(LitElement) {
   render() {
      return html`
         <div>
            ...
            <a href="..." class="d2l-arrowkeys-focusable">link 1</a>
            <a href="..." class="d2l-arrowkeys-focusable">link 2</a>
            ...
         </div>
      `;
   }
}
customElements.define('my-element', MyElement);
```
> If the elements are not known up front, or the elements cannot be simply queried in the element's local DOM-scope, an `async arrowKeysFocusablesProvider` may be implemented.
```javascript
import { ArrowKeysMixin } from '@brightspace-ui/core/mixins/arrowkeys-mixin.js';
class MyElement extends ArrowKeysMixin(LitElement) {
   render() {
      return html`
         <div>
            ...
            <a href="...">link 1</a>
            <a href="...">link 2</a>
            ...
         </div>
      `;
   }
   async arrowKeysFocusablesProvider() {
      return [ /* array containing focusable elements */]
   }
}
customElements.define('my-element', MyElement);
```
Lastly, we can include a properties section that lists the optional properties.

***Properties:***

* `arrowKeysDirection` (optional): The RTL behavior (default is `leftright`)
* `arrowKeysNoWrap` (optional): To opt out of wrapping focus from end-to-start and start-to-end
* `arrowKeysBeforeFocus` (optional): Async callback invoked before focus us applied